src/components/ManualMapPoints.js

```diff
@@ -0,0 +1,141 @@
+import React, { useState, useEffect } from 'react';
+import { MapContainer, Marker, TileLayer, useMapEvents } from 'react-leaflet';
+import Shapes from 'components/Shapes';
+import Points from 'components/Points';
+import useInitalLocation from 'logic/useInitalLocation';
+import { icons, mapType } from 'logic/TypeLogic';
+import { addPointToMap } from 'logic/FbPointLogic';
+
+function toIsoString(date) {
+  const pad = function(num) {
+    return (num < 10 ? '0' : '') + num;
+  };
+
+  return date.getFullYear() +
+    '-' + pad(date.getMonth() + 1) +
+    '-' + pad(date.getDate()) +
+    'T' + pad(date.getHours()) +
+    ':' + pad(date.getMinutes()) +
+    ':' + pad(date.getSeconds());
+}
+
+export default function ManualMapPoints() {
+  const [screenDatabase, setScreenDatabase] = useState('intruder-points');
+  const [position] = useInitalLocation(screenDatabase);
+  const [key, setKey] = useState('');
+  const [clickedPos, setClickedPos] = useState({ lat: 0, lng: 0 });
+  const [pointType, setPointType] = useState('');
+  const [pointName, setPointName] = useState('');
+  const [pointTimeStamp, setPointTimeStamp] = useState(toIsoString(new Date()));
+  const [timeUpdate, setTimeUpdate] = useState(false);
+  const [img64, setImg64] = useState('');
+
+  function MapEvents() {
+    useMapEvents({
+      click(e) {
+        setClickedPos({ lat: e.latlng.lat, lng: e.latlng.lng });
+      }
+    });
+  }
+
+  const onChangeDb = (e) => {
+    setScreenDatabase(e.target.value);
+    setKey('key_' + e.target.value);
+  };
+
+  const onChangeLatLng = (type, e) => {
+    if (isNaN(e.target.value)) { return; }
+    const x = { ...clickedPos };
+    x[type] = e.target.value;
+    setClickedPos(x);
+  };
+
+  const onChangeDateTime = (e) => {
+    const date = new Date(e.target.value);
+    setPointTimeStamp(date);
+  };
+
+  const onCheckedAutoTime = () => {
+    setTimeUpdate(!timeUpdate);
+  };
+
+  const onSubmit = () => {
+    addPointToMap(screenDatabase, { img: img64, name: pointName, type: pointType, shooted: false }, clickedPos, pointTimeStamp);
+  };
+
+  useEffect(() => {
+    if (!timeUpdate) { return; }
+    const interFunction = () => {
+      setPointTimeStamp(toIsoString(new Date()));
+    };
+    interFunction();
+    const interval = setInterval(interFunction, 500);
+    return () => clearInterval(interval);
+  }, [timeUpdate]);
```

Comment:
To chyba jest zbędne. Wystarczy w onSubmit pobrać czas jeśli timeUpdate true.

Reply:
Tak tak, to miał być wizualny bajer żebyś wiedział ze czas sie aktualizuje w polu więc będzie aktualny w bazie